### PR TITLE
doc(qa): Initial versions of Metadata Service test plan

### PR DESCRIPTION
**Pull request by themarcelor**
This initial versions describes some basic coverage. TBD: Negative tests

### CODE REVIEW

File: docs/mds_test_plan.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+2. Get existing keys from metadata records - positive test
+3. Query metadata for matching records - positive test
+
+### Load Tests
+
+#### Performance benchmarking
+The service is currently being load tested with the following criteria:
+
+#### Soak test
+This test involves a continuous 4hs run, constantly creating random entries and running queries with 200 virtual users.
```
Comment (fantix) [inline docs/mds_test_plan.md]: Codacy Issue found: [[no-consecutive-blank-lines] Remove 1 line after node](https://app.codacy.com/gh/uc-cdis/metadata-service/pullRequest?prid=5189388)

**Comment by fantix** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 8
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/uc-cdis/metadata-service/pullRequest?prid=5189388&bid=17024187)

File: docs/mds_test_plan.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+{"foo":123}
+```
+
+### Tests
+
+The initial coverage comprises the following scenarios:
+
+1. Create records against the metadata svc blobstore - positive test
+2. Get existing keys from metadata records - positive test
+3. Query metadata for matching records - positive test
```
Comment (fantix) [inline docs/mds_test_plan.md]: Codacy Issue found: [[list-item-indent] Incorrect list-item indent: add 1 space](https://app.codacy.com/gh/uc-cdis/metadata-service/pullRequest?prid=5189388)

File: docs/mds_test_plan.md [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+% curl --location --request POST 'https://${GEN3_HOSTNAME}/mds/metadata/someGUID2' \
+--header 'Content-Type: application/json' \
+--header "Authorization: Basic ${BASIC_AUTH_MDS}" \
+--data-raw '{
+        "bar": 789
+}'
+{"bar":789}
+```
+Or with a Gen3 Access Token, e.g.:
+```
```
Comment (fantix) [inline docs/mds_test_plan.md]: Codacy Issue found: [[fenced-code-flag] Missing code-language flag](https://app.codacy.com/gh/uc-cdis/metadata-service/pullRequest?prid=5189388)

File: docs/mds_test_plan.md [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+--data-raw '{ "foo": 123 }'
+{"foo":123}
+```
+
+### Tests
+
+The initial coverage comprises the following scenarios:
+
+1. Create records against the metadata svc blobstore - positive test
+2. Get existing keys from metadata records - positive test
```
Comment (fantix) [inline docs/mds_test_plan.md]: Codacy Issue found: [[list-item-indent] Incorrect list-item indent: add 1 space](https://app.codacy.com/gh/uc-cdis/metadata-service/pullRequest?prid=5189388)

File: docs/mds_test_plan.md [hunk 5]
```diff
@@ -0,0 +1,46 @@
+# Metadata Service (mds) test plan
```
Comment (fantix) [inline docs/mds_test_plan.md]: Codacy Issue found: [[no-file-name-irregular-characters] Do not use `_` in a file name](https://app.codacy.com/gh/uc-cdis/metadata-service/pullRequest?prid=5189388)

File: docs/mds_test_plan.md [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+
+### One-liner overview of the service
+The MDS is a blobstore used to store JSON-formatted data, any schemaless data.
+
+### RESTful API
+Here's one of its swagger pages:
+https://nci-crdc-staging.datacommons.io/mds/docs
```
Comment (fantix) [inline docs/mds_test_plan.md]: Codacy Issue found: [[no-literal-urls] Don’t use literal URLs without angle brackets](https://app.codacy.com/gh/uc-cdis/metadata-service/pullRequest?prid=5189388)

File: docs/mds_test_plan.md [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+--header "Authorization: Bearer ${ACCESS_TOKEN}" \
+--data-raw '{ "foo": 123 }'
+{"foo":123}
+```
+
+### Tests
+
+The initial coverage comprises the following scenarios:
+
+1. Create records against the metadata svc blobstore - positive test
```
Comment (fantix) [inline docs/mds_test_plan.md]: Codacy Issue found: [[list-item-indent] Incorrect list-item indent: add 1 space](https://app.codacy.com/gh/uc-cdis/metadata-service/pullRequest?prid=5189388)

File: docs/mds_test_plan.md [hunk 8]
```diff
@@ -0,0 +1,46 @@
+# Metadata Service (mds) test plan
+
+### One-liner overview of the service
```
Comment (fantix) [inline docs/mds_test_plan.md]: Codacy Issue found: [[heading-increment] Heading levels should increment by one level at a time](https://app.codacy.com/gh/uc-cdis/metadata-service/pullRequest?prid=5189388)

**Comment by fantix** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 8
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/uc-cdis/metadata-service/pullRequest?prid=5189388&bid=17024187)

File: docs/mds_test_plan.md [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+### Tests
+The initial coverage comprises the following scenarios:
+1. Create records against the metadata svc blobstore - positive test
+2. Get existing keys from metadata records - positive test
+3. Query metadata for matching records - positive test
+### Load Tests
+#### Performance benchmarking
+The service is currently being load tested with the following criteria:
+    http_req_duration: ['avg<3000', 'p(95)<15000'],
+    failed_requests: ['rate<0.1'],
```
Comment (fantix) [inline docs/mds_test_plan.md]: Codacy Issue found: [[no-shortcut-reference-link] Use the trailing [] on reference links](https://app.codacy.com/gh/uc-cdis/metadata-service/pullRequest?prid=5189388)

File: docs/mds_test_plan.md [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+```
+### Tests
+The initial coverage comprises the following scenarios:
+1. Create records against the metadata svc blobstore - positive test
+2. Get existing keys from metadata records - positive test
+3. Query metadata for matching records - positive test
+### Load Tests
+#### Performance benchmarking
+The service is currently being load tested with the following criteria:
+    http_req_duration: ['avg<3000', 'p(95)<15000'],
```
Comment (fantix) [inline docs/mds_test_plan.md]: Codacy Issue found: [[no-shortcut-reference-link] Use the trailing [] on reference links](https://app.codacy.com/gh/uc-cdis/metadata-service/pullRequest?prid=5189388)

File: docs/mds_test_plan.md [hunk 11]
```diff
@@ -0,0 +1,39 @@
+# Metadata Service (mds) test plan
+
+### One-liner overview of the service
```
Comment (fantix) [inline docs/mds_test_plan.md]: Codacy Issue found: [[heading-increment] Heading levels should increment by one level at a time](https://app.codacy.com/gh/uc-cdis/metadata-service/pullRequest?prid=5189388)

File: docs/mds_test_plan.md [hunk 12]
```diff
@@ -0,0 +1,39 @@
+# Metadata Service (mds) test plan
```
Comment (fantix) [inline docs/mds_test_plan.md]: Codacy Issue found: [[no-file-name-irregular-characters] Do not use `_` in a file name](https://app.codacy.com/gh/uc-cdis/metadata-service/pullRequest?prid=5189388)

File: docs/mds_test_plan.md [hunk 13]
```diff
@@ -0,0 +1,39 @@
...
+% curl --location --request POST 'https://${GEN3_HOSTNAME}/mds/metadata/someGUID2' \
+--header 'Content-Type: application/json' \
+--header "Authorization: Basic ${BASIC_AUTH_MDS}" \
+--data-raw '{
+        "bar": 789
+}'
+{"bar":789}
+```
+Or with a Gen3 Access Token, e.g.:
+```
```
Comment (fantix) [inline docs/mds_test_plan.md]: Codacy Issue found: [[fenced-code-flag] Missing code-language flag](https://app.codacy.com/gh/uc-cdis/metadata-service/pullRequest?prid=5189388)

File: docs/mds_test_plan.md (hunk 9 above)
Comment (fantix) [inline docs/mds_test_plan.md]: Codacy Issue found: [[no-undefined-references] Found reference to undefined definition](https://app.codacy.com/gh/uc-cdis/metadata-service/pullRequest?prid=5189388)

File: docs/mds_test_plan.md [hunk 14]
```diff
@@ -0,0 +1,39 @@
+# Metadata Service (mds) test plan
+
+### One-liner overview of the service
+The MDS is a blobstore used to store JSON-formatted data, any schemaless data.
+### RESTful API
+Here's one of its swagger pages:
+https://nci-crdc-staging.datacommons.io/mds/docs
```
Comment (fantix) [inline docs/mds_test_plan.md]: Codacy Issue found: [[no-literal-urls] Don’t use literal URLs without angle brackets](https://app.codacy.com/gh/uc-cdis/metadata-service/pullRequest?prid=5189388)

File: docs/mds_test_plan.md (hunk 10 above)
Comment (fantix) [inline docs/mds_test_plan.md]: Codacy Issue found: [[no-undefined-references] Found reference to undefined definition](https://app.codacy.com/gh/uc-cdis/metadata-service/pullRequest?prid=5189388)

**Comment by fantix** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 3
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/uc-cdis/metadata-service/pullRequest?prid=5189388&bid=17024187)

File: docs/mds-test-plan.md [hunk 15]
```diff
@@ -0,0 +1,39 @@
...
+## Tests
+The initial coverage comprises the following scenarios:
+1. Create records against the metadata svc blobstore - positive test
+2. Get existing keys from metadata records - positive test
+3. Query metadata for matching records - positive test
+## Load Tests
+### Performance benchmarking
+The service is currently being load tested with the following criteria:
+* http_req_duration: avg<1000, p(95)<2000
+* failed_requests: rate<0.05
```
Comment (fantix) [inline docs/mds-test-plan.md]: Codacy Issue found: [[list-item-indent] Incorrect list-item indent: add 2 spaces](https://app.codacy.com/gh/uc-cdis/metadata-service/pullRequest?prid=5189388)

File: docs/mds-test-plan.md [hunk 16]
```diff
@@ -0,0 +1,39 @@
...
+```
+## Tests
+The initial coverage comprises the following scenarios:
+1. Create records against the metadata svc blobstore - positive test
+2. Get existing keys from metadata records - positive test
+3. Query metadata for matching records - positive test
+## Load Tests
+### Performance benchmarking
+The service is currently being load tested with the following criteria:
+* http_req_duration: avg<1000, p(95)<2000
```
Comment (fantix) [inline docs/mds-test-plan.md]: Codacy Issue found: [[list-item-indent] Incorrect list-item indent: add 2 spaces](https://app.codacy.com/gh/uc-cdis/metadata-service/pullRequest?prid=5189388)

File: docs/mds-test-plan.md [hunk 17]
```diff
@@ -0,0 +1,39 @@
...
+* failed_requests: rate<0.05
+* virtual_users: <ramping up from 0 to 50 users in ~5 mins and then bumping up to 100 users in a short time-frame (10s)> and keep running for 2 more minutes>
```
Comment (fantix) [inline docs/mds-test-plan.md]: Codacy Issue found: [[list-item-indent] Incorrect list-item indent: add 2 spaces](https://app.codacy.com/gh/uc-cdis/metadata-service/pullRequest?prid=5189388)

**Comment by fantix** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 3
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/uc-cdis/metadata-service/pullRequest?prid=5189388&bid=17024187)

File: docs/mds-test-plan.md [hunk 18]
```diff
@@ -0,0 +1,39 @@
...
+## Tests
+The initial coverage comprises the following scenarios:
+1. Create records against the metadata svc blobstore - positive test
+2. Get existing keys from metadata records - positive test
+3. Query metadata for matching records - positive test
+## Load Tests
+### Performance benchmarking
+The service is currently being load tested with the following criteria:
+* http_req_duration: avg<1000, p(95)<2000
+* failed_requests: rate<0.05
```
Comment (fantix) [inline docs/mds-test-plan.md]: Codacy Issue found: [[list-item-indent] Incorrect list-item indent: add 2 spaces](https://app.codacy.com/gh/uc-cdis/metadata-service/pullRequest?prid=5189388)

File: docs/mds-test-plan.md [hunk 19]
```diff
@@ -0,0 +1,39 @@
...
+```
+## Tests
+The initial coverage comprises the following scenarios:
+1. Create records against the metadata svc blobstore - positive test
+2. Get existing keys from metadata records - positive test
+3. Query metadata for matching records - positive test
+## Load Tests
+### Performance benchmarking
+The service is currently being load tested with the following criteria:
+* http_req_duration: avg<1000, p(95)<2000
```
Comment (fantix) [inline docs/mds-test-plan.md]: Codacy Issue found: [[list-item-indent] Incorrect list-item indent: add 2 spaces](https://app.codacy.com/gh/uc-cdis/metadata-service/pullRequest?prid=5189388)

File: docs/mds-test-plan.md [hunk 20]
```diff
@@ -0,0 +1,39 @@
...
+* failed_requests: rate<0.05
+* virtual_users: <ramping up from 0 to 50 users in ~5 mins and then bumping up to 100 users in a short time-frame (10s)> and keep running for 2 more minutes>
```
Comment (fantix) [inline docs/mds-test-plan.md]: Codacy Issue found: [[list-item-indent] Incorrect list-item indent: add 2 spaces](https://app.codacy.com/gh/uc-cdis/metadata-service/pullRequest?prid=5189388)

**Comment by fantix** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 3
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/uc-cdis/metadata-service/pullRequest?prid=5189388&bid=17024187)

File: docs/mds-test-plan.md [hunk 21]
```diff
@@ -0,0 +1,39 @@
...
+The initial coverage comprises the following scenarios:
+1. Create records against the metadata svc blobstore - positive test
+2. Get existing keys from metadata records - positive test
+3. Query metadata for matching records - positive test
+## Load Tests
+### Performance benchmarking
+The service is currently being load tested with the following criteria:
+  * http_req_duration: avg<1000, p(95)<2000
+  * failed_requests: rate<0.05
+  * virtual_users: <ramping up from 0 to 50 users in ~5 mins and then bumping up to 100 users in a short time-frame (10s)> and keep running for 2 more minutes>
```
Comment (fantix) [inline docs/mds-test-plan.md]: Codacy Issue found: [[list-item-bullet-indent] Incorrect indentation before bullet: remove 2 spaces](https://app.codacy.com/gh/uc-cdis/metadata-service/pullRequest?prid=5189388)

File: docs/mds-test-plan.md [hunk 22]
```diff
@@ -0,0 +1,39 @@
...
+## Tests
+The initial coverage comprises the following scenarios:
+1. Create records against the metadata svc blobstore - positive test
+2. Get existing keys from metadata records - positive test
+3. Query metadata for matching records - positive test
+## Load Tests
+### Performance benchmarking
+The service is currently being load tested with the following criteria:
+  * http_req_duration: avg<1000, p(95)<2000
+  * failed_requests: rate<0.05
```
Comment (fantix) [inline docs/mds-test-plan.md]: Codacy Issue found: [[list-item-bullet-indent] Incorrect indentation before bullet: remove 2 spaces](https://app.codacy.com/gh/uc-cdis/metadata-service/pullRequest?prid=5189388)

File: docs/mds-test-plan.md [hunk 23]
```diff
@@ -0,0 +1,39 @@
...
+```
+## Tests
+The initial coverage comprises the following scenarios:
+1. Create records against the metadata svc blobstore - positive test
+2. Get existing keys from metadata records - positive test
+3. Query metadata for matching records - positive test
+## Load Tests
+### Performance benchmarking
+The service is currently being load tested with the following criteria:
+  * http_req_duration: avg<1000, p(95)<2000
```
Comment (fantix) [inline docs/mds-test-plan.md]: Codacy Issue found: [[list-item-bullet-indent] Incorrect indentation before bullet: remove 2 spaces](https://app.codacy.com/gh/uc-cdis/metadata-service/pullRequest?prid=5189388)

**Comment by fantix** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 3
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/uc-cdis/metadata-service/pullRequest?prid=5189388&bid=17024187)

File: docs/mds-test-plan.md (hunk 18 above)
Comment (fantix) [inline docs/mds-test-plan.md]: Codacy Issue found: [[list-item-indent] Incorrect list-item indent: add 2 spaces](https://app.codacy.com/gh/uc-cdis/metadata-service/pullRequest?prid=5189388)

File: docs/mds-test-plan.md (hunk 19 above)
Comment (fantix) [inline docs/mds-test-plan.md]: Codacy Issue found: [[list-item-indent] Incorrect list-item indent: add 2 spaces](https://app.codacy.com/gh/uc-cdis/metadata-service/pullRequest?prid=5189388)

File: docs/mds-test-plan.md (hunk 20 above)
Comment (fantix) [inline docs/mds-test-plan.md]: Codacy Issue found: [[list-item-indent] Incorrect list-item indent: add 2 spaces](https://app.codacy.com/gh/uc-cdis/metadata-service/pullRequest?prid=5189388)

**Comment by haraprasadj** [comment]
Should we not store the test plans in gen3-qa?

**Comment by themarcelor** [comment]
Closing this PR to move this test plan to `gen3-qa` as per @haraprasadj 's comment.

**Pull request closed without merging**